src/LineageTree/lineageTree.py

```diff
@@ -2400,6 +2403,445 @@ def get_cells_at_t_from_root(self, r: [int, list], t: int = None) -> list:
+        im = ax.imshow(
+            cost_mat, cmap="viridis", origin="lower", interpolation="nearest"
+        )
+        plt.colorbar(im)
+        ax.set_title("Heatmap of DTW Cost Matrix")
+        ax.set_xlabel("Tree 1")
+        ax.set_ylabel("tree 2")
+        x_path, y_path = zip(*path)
+        ax.plot(y_path, x_path, color="black")
+
+        return cost, fig
+
+    def plot_dtw_trajectory(
+        self,
+        nodes1: int,
+        nodes2: int,
+        threshold: int=1000,
+        regist: bool=True,
+        start_d: int=0,
+        back_d: int=0,
+        fast: bool=False,
+        w: int=0,
+        centered_band: bool=True,
+        three_dim: bool=True,
+        alig: bool=False,
+    ) -> (float, plt.figure):
+        """
+        Plots DTW trajectories aligment between two cell cycles in 2D or 3D
+
+            Args:
+                nodes1 (int): node to compare distance
+                nodes2 (int): node to compare distance
+                threshold (int): set a maximum number of points a track can have
+                regist (boolean): Rotate and translate trajectories
+                start_d (int): start delay
+                back_d (int): end delay
+                w (int): window size
+                fast (boolean): True if the user wants to run the fast algorithm with window restrains
+                centered_band (boolean): if running the fast algorithm, True if the windown is centered
+                three_dim (boolean): True if plot is 3D, else 2D
+                alig (boolean): True to show alignment on plot
+
+            Returns:
+                (float) DTW distance
+                (figue) Trajectories Plot
+        """
+        distance, alignment, cost_mat, pos_cycle1, pos_cycle2 = self.calculate_dtw(
+            nodes1,
+            nodes2,
+            threshold,
+            regist,
+            start_d,
+            back_d,
+            fast,
+            w,
+            centered_band,
+            cost_mat_p=True,
+        )
+
+        fig = plt.figure(figsize=(10, 6))
+        if three_dim:
+            ax = fig.add_subplot(1, 1, 1, projection="3d")
+        else:
+            ax = fig.add_subplot(1, 1, 1)
+
+        if three_dim:
+            ax.plot(
+                pos_cycle1[:, 0],
+                pos_cycle1[:, 1],
+                pos_cycle1[:, 2],
+                "-",
+                label=f"root = {nodes1}",
+            )
+            ax.plot(
+                pos_cycle2[:, 0],
+                pos_cycle2[:, 1],
+                pos_cycle2[:, 2],
+                "-",
+                label=f"root = {nodes2}",
+            )
+        else:
+            ax.plot(
+                pos_cycle1[:, 0],
+                pos_cycle1[:, 1],
+                "-",
+                label=f"root = {nodes1}",
+            )
+            ax.plot(
+                pos_cycle2[:, 0],
+                pos_cycle2[:, 1],
+                "-",
+                label=f"root = {nodes2}",
+            )
+
```

Comment:
I don't know how to best import this library

Reply:
which library?

Reply:
from sklearn.decomposition import PCA

Reply:
Because it's a large library and it's only needed there, you can import it within the function.

You should then put a warning message if they want to use pca but don't have sklearn.

To check if they have it you can do
```python
try:
    from sklearn.decomposition import PCA
except ImportError:
    Warning("scikit-learn is not installed, the PCA orientation cannot be used. You can install scikit-learn with pip install")
```
Or something like that.